metals/src/main/scala/scala/meta/internal/metals/codelenses/SuperMethodLensesProvider.scala

```diff
@@ -0,0 +1,126 @@
+package scala.meta.internal.metals.codelenses
+
+import java.util.Collections.singletonList
+import org.eclipse.{lsp4j => l}
+import scala.collection.{mutable => m}
+import scala.meta.internal.implementation.GlobalClassTable
+import scala.meta.internal.implementation.ImplementationProvider
+import scala.meta.internal.implementation.SuperMethodProvider
+import scala.meta.internal.implementation.SymbolWithAsSeenFrom
+import scala.meta.internal.implementation.TextDocumentWithPath
+import scala.meta.internal.metals.Buffers
+import scala.meta.internal.metals.BuildTargets
+import scala.meta.internal.metals.MetalsServerConfig
+import scala.meta.internal.metals.ServerCommands
+import scala.meta.internal.metals.TokenEditDistance
+import scala.meta.internal.metals.UserConfiguration
+import scala.meta.internal.semanticdb.SymbolInformation
+import scala.meta.internal.semanticdb.SymbolOccurrence
+import scala.meta.internal.semanticdb.TextDocument
+import scala.meta.internal.symtab.GlobalSymbolTable
+import scala.meta.io.AbsolutePath
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.codelenses.SuperMethodLensesProvider.LensGoSuperCache
+import scala.meta.internal.metals.codelenses.SuperMethodLensesProvider.emptyLensGoSuperCache
+
+final class SuperMethodLensesProvider(
+    implementationProvider: ImplementationProvider,
+    buffers: Buffers,
+    buildTargets: BuildTargets,
+    userConfig: () => UserConfiguration,
+    config: MetalsServerConfig
+) extends CodeLenses {
+
+  override def isEnabled: Boolean = userConfig().superMethodLensesEnabled
+
+  override def codeLenses(
+      textDocumentWithPath: TextDocumentWithPath
+  ): Seq[l.CodeLens] = {
+    val textDocument = textDocumentWithPath.textDocument
+    val path = textDocumentWithPath.filePath
+
+    val search =
+      makeSymbolSearchMethod(
+        makeGlobalClassTable(textDocumentWithPath.filePath),
+        textDocumentWithPath.textDocument
+      )
+    val distance =
+      TokenEditDistance.fromBuffer(path, textDocument.text, buffers)
+
+    for {
+      occurrence <- textDocument.occurrences
+      if occurrence.role.isDefinition
+      symbol = occurrence.symbol
+      gotoSuperMethod <- createSuperMethodCommand(
+        textDocumentWithPath,
+        symbol,
+        occurrence.role,
+        emptyLensGoSuperCache(),
+        search
+      ).toIterable
+      range <- occurrence.range
+        .flatMap(r => distance.toRevised(r.toLSP))
+        .toList
+    } yield new l.CodeLens(range, gotoSuperMethod, null)
+  }
+
+  private def createSuperMethodCommand(
+      docWithPath: TextDocumentWithPath,
+      symbol: String,
+      role: SymbolOccurrence.Role,
+      cache: LensGoSuperCache,
+      findSymbol: String => Option[SymbolInformation]
+  ): Option[l.Command] = {
+    if (userConfig().superMethodLensesEnabled) {
```

Comment:
removed this as `enabled` is checked in other place, this is old check which I forgot to remove